client/src/ui/BitbakeTerminalLinkProvider.ts

```diff
@@ -0,0 +1,48 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode'
+import { type BitBakeProjectScanner } from '../driver/BitBakeProjectScanner'
+import * as fs from 'fs'
+
+export class BitbakeTerminalLinkProvider implements vscode.TerminalLinkProvider {
+  bitBakeProjectScanner: BitBakeProjectScanner
+
+  constructor (bitBakeProjectScanner: BitBakeProjectScanner) {
+    this.bitBakeProjectScanner = bitBakeProjectScanner
+  }
+
+  async provideTerminalLinks (context: vscode.TerminalLinkContext, token: vscode.CancellationToken): Promise<vscode.TerminalLink[]> {
+    const links: vscode.TerminalLink[] = []
+    // We only have additional links to provide if we need to resolve container paths
+    if (!this.bitBakeProjectScanner.needsContainerPathsResolution()) { return links }
+
+    // Match any string that starts with a slash enclosed in spaces or special characters
+    const regex = /(^|[^a-zA-Z0-9_/.-])(\/[a-zA-Z0-9_/.-]+)(?=$|[^a-zA-Z0-9_/.-])/g
```

Comment:
In my understanding, this regex is supposed to match paths. I think there might be several issues

1. It does not match `./something`
2. A valid path could contains many "special characters".
3. It does not ensure the opening "special character" is the same as the "closing" one. If the path is `"/some*path/"`, then it will match `/some`. I don't think a regex can detect it. Well, it could with "back references", but it won't work here because of `^` and `$`.

Reply:
1. Almost all paths reported by bitbake in the terminal use absolute paths. Here's an example log:
```log
NOTE: Running setscene task 307 of 369 (/home/deribaucourt/Workspace/yocto-vscode/yocto/yocto-build/sources/poky/meta/recipes-support/libcheck/libcheck_0.15.2.bb:do_deploy_source_date_epoch_setscene)
NOTE: Running setscene task 308 of 369 (/home/deribaucourt/Workspace/yocto-vscode/yocto/yocto-build/sources/poky/meta/recipes-support/libnl/libnl_3.9.0.bb:do_deploy_source_date_epoch_setscene)
Parsing recipes...ERROR: ParseError at /home/deribaucourt/Workspace/yocto-vscode/yocto/yocto-build/sources/poky/meta/recipes-core/busybox/busybox_1.36.1.bb:58: unparsed line: 'undefinedvar'
Summary: There were 2 ERROR messages, returning a non-zero exit code.
Traceback (most recent call last):
  File "/home/deribaucourt/Workspace/yocto-vscode/yocto/yocto-build/sources/poky/bitbake/bin/bitbake-layers", line 95, in <module>
    ret = main()
```
2. "special characters" other than `/-._+` are usually not allowed in paths or very rare. But this also doesn't support localized characters like accents or foreign alphabets. It works for most standard setups.
3. I don't think it mattersto check the correspondance, I only want to provide links in paths
Even though this feature is not 100% accurate, it provides great results when using KAS for instance. It's a bonus feature to fix some of the paths used by docker containers. It's not critical if it misses some configurations. I'd like to merge this feature knowing these limitations. If a user needs more advanced links, they can contribute an improvment.